hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/codec/TokenKeyWrapper.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.store.contracts.precompile.codec;
+
+public record TokenKeyWrapper(int keyType, KeyValueWrapper key) {
+    public boolean isUsedForAdminKey() {
```

Comment:
Should we add an "isValid" method that returns `Integer.bitCount(keyType) == 1`?  It would be useful in some of the validation logic elsewhere.

Reply:
I'm not sure if I understood you properly, but the "Integer.bitCount(keyType) == 1"  is  not a suitable option. For example, in CreatePrecompileSuite.fungibleTokenCreateHappyPath() test KYC reuses the admin key's ecdsa[] and more than one of the bits is significant.

Reply:
Perhaps 'isValid' is the wrong name, perhaps 'isSingleType', and then the logic here https://github.com/hashgraph/hedera-services/pull/3738/files#diff-0f30f19273b290c73c5b9f6116ec5afc47c071f2e4f0fe24828ef1ff7063ba22R109 would have a lot less checks.

Reply:
Aren't we mixing `KeyType` with `KeyValue` types?

We have 7 types of key types and 5 types of key values. I'm not sure how we would use this `isSingleType` method in setting the key value type. For setting the correct value we should check what value type has been set - `shouldInheritAccountKey`, `contractID`, `ed25519`, `ecdsaSecp256k1`, `delegatableContractID`.

Reply:
Anyway, I suggest to apply potential improvements for the key logic/checks in a separate PR, as we'll deal with extracting common logic from `TokenUpdateTransitionLogic` and `TokenUpdateLogic` in a separate PR.